lib/_http_client.js

```diff
@@ -115,7 +115,7 @@ function ClientRequest(options, cb) {
   var host = options.host = validateHost(options.hostname, 'hostname') ||
                             validateHost(options.host, 'host') || 'localhost';
 
-  var setHost = (options.setHost === undefined);
+  var setHost = (options.setHost === undefined || options.setHost);
```

Comment:
Maybe just `Boolean(options.setHost)`? 

Also this is technically a breaking (semver) change.

Reply:
How about `!!options.setHost`?

Reply:
I prefer `Boolean` since it's more explicit (we are converting this to a boolean) rather than "we are negating this, which is coercing it to a boolean, and then negating it again to get the boolean value".

Reply:
Using only `Boolean(options.setHost)` is not ok as `undefined` set `setHost` to `true`.

Reply:
Doesn't it a bugfix but breaking?

Reply:
Technically yes but Imho it's semver-patch as I doubt anyone is relying on the current behavior. The option is currently not documented.